src/main/java/com/google/acai/TestScope.java

```diff
@@ -16,31 +16,33 @@
 
 package com.google.acai;
 
+import static com.google.common.base.Preconditions.checkState;
+
 import com.google.inject.AbstractModule;
 import com.google.inject.Key;
 import com.google.inject.OutOfScopeException;
 import com.google.inject.Provider;
 import com.google.inject.Scope;
 
-import java.util.HashMap;
 import java.util.Map;
-
-import static com.google.common.base.Preconditions.checkState;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.atomic.AtomicReference;
 
 /**
  * Scope for bindings annotated with {@link TestScoped}.
  */
 class TestScope implements Scope {
-  private final ThreadLocal<Map<Key<?>, Object>> values = new ThreadLocal<>();
+  private AtomicReference<ConcurrentMap<Key<?>, Object>> values = new AtomicReference<>();
```

Comment:
The previous behavior allowed for test runners which execute multiple tests concurrently. I think that will no longer work with this change.

I added a test in commit 3c39c2461cc8212f87915090208a8c365c60d6ca to clarify the existing behavior. I think if a single test case uses multiple threads it may be best for that test just to get the instances it needs on the main test thread and then pass those to the other threads.

What do you think? Is there another option which does what you want while keeping the support for parallel test runners? Can you add a test case which motivates this change to clarify what it enables?